### PR TITLE
mm/validate: fully migrate and clear bitmap & other cleanups

### PR DESCRIPTION
**Pull request by 00xc**
* Remove useless `pub` on `ValidBitmap` methods.
* Fix: if the number of pages in the bitmap was not a multiple of 64, the remainder would not be copied over to the new bitmap during migration, which could lead to tracking validated pages as invalid.
* Use an `usize` to index into the bitmap, as offsets are always positive.
* Fully clear the bitmap after initialization.

### CODE REVIEW

File: src/mm/validate.rs [hunk 1]
```diff
@@ -152,11 +152,20 @@ impl ValidBitmap {
...
 
-    fn migrate(&mut self, new_bitmap: *mut u64) {
-        let (count, _) = self.index(self.pend);
+    /// The number of u64's in the bitmap
+    fn bitmap_len(&self) -> usize {
+        let num_pages = self.pend.bits().saturating_sub(self.pbase.bits()) / PAGE_SIZE;
+        let mut num_elems = num_pages / u64::BITS as usize;
+        if num_elems % u64::BITS as usize != 0 {
+            num_elems += 1;
+        }
```
Comment (Freax13) [inline src/mm/validate.rs:156]: Is the check correct? I'd expect it to be `num_pages % u64::BITS as usize != 0`. Note that you may be able to use [`usize::div_ceil`](https://doc.rust-lang.org/std/primitive.usize.html#method.div_ceil) here (which was stabilized very recently).

Comment (00xc) [reply src/mm/validate.rs:156]: Yep, nice catch, fixed in the new version with `div_ceil()`

**Review by Freax13 — APPROVED** [review]

**Comment by 00xc** [comment]
@joergroedel I think this is ready to be merged

**Comment by joergroedel** [comment]
@00xc This breaks `make test-in-svsm`. Can you please have a look?

**Comment by joergroedel** [comment]
Bisection points at

```
commit 9e5d903b2d940bf2e81cfbf305582af7e35591fd (HEAD)
Author: Carlos López <carlos.lopez@suse.com>
Date:   Tue Oct 17 14:35:44 2023 +0200

    mm/validate: use usize instead of isize to index into the bitmap
    
    The offsets are always positive from the start of the bitmap, so use
    an unsigned type.
    
    Signed-off-by: Carlos López <carlos.lopez@suse.com>
```

**Comment by 00xc** [comment]
> 9e5d903b2d940bf2e81cfbf305582af7e35591fd

Only happens in debug builds, so I suspect some debug assertion is being triggered.

**Comment by 00xc** [comment]
I'm working on a bigger rework of the bitmap to increase safety.

**Comment by 00xc** [comment]
Turns out my assumption on 498b599221f0fa7b9f7eed073ebcfbea1f5690aa was wrong, things should work now. I'll push my rework of the bitmap to a separate PR.

**Pull request merged**